dissect/target/plugins/scrape/scrape.py

```diff
@@ -204,3 +286,96 @@ def scrape_needles_from_disks(
             disk.seek(0)
             for needle, offset, match in find_needles(disk, needles=needles, block_size=block_size):
                 yield disk, needle, offset, match
+
+    @internal
+    def _get_disk_vols(self, disk: Container) -> Iterator[Volume]:
+        """Yields all volumes for a given disk container.
+
+        When the disk has an associated volume system, volumes are retrieved from there.
+        When the disk is raw, volumes are retrieved from the target's volume list.
+
+        Args:
+            disk: The disk container to get volumes for.
+        """
+        if disk.vs:
+            yield from sorted(disk.vs.volumes, key=lambda v: v.offset)
+        else:
+            for volume in sorted(self.target.volumes, key=lambda v: v.offset or 0):
+                if volume.disk == disk and not volume.vs:
+                    yield volume
+
+
+def process_lvm_layer(context: ScrapeContext, volume: Volume) -> bool:
+    if not (context.lvm_flag and isinstance(volume.vs, LogicalVolumeSystem)):
+        return False
+
+    dependencies_met = True
+    backing_regions_info = []
+
+    try:
+        for source_dev in volume.vs.fh:
+            backing_vol_obj = source_dev.fh
+
+            if backing_vol_obj not in context.volume_region_map:
+                dependencies_met = False
+                break
+
+            backing_regions_info.append(context.volume_region_map[backing_vol_obj])
+
+    except (AttributeError, TypeError):
+        dependencies_met = False
+
+    if dependencies_met:
+        if not context.all_flag:
+            for map_key, region_key in backing_regions_info:
+                # Only delete the scrape_map region if it hasn't
+                # been deleted by another LV in this VG.
+                # This occurs when multiple LVs share the same backing PV.
+                key_to_delete = (map_key, region_key)
+                if key_to_delete not in context.deleted_scrape_regions:
+                    del context.scrape_map[map_key][region_key]
+                    context.deleted_scrape_regions.add(key_to_delete)
+
+        # Add the new LVM volume as a new "disk"
+        new_region_key = (0, volume.size)
+        context.scrape_map[volume][new_region_key] = (volume, 0)
+
+        # Add this new LV to our index
+        context.volume_region_map[volume] = (volume, new_region_key)
+        return True
+    return False
+
+
+def process_encrypted_layer(context: ScrapeContext, volume: Volume) -> bool:
+    if not (context.encrypted_flag and isinstance(volume.vs, EncryptedVolumeSystem)):
+        return False
+
+    backing_vol_obj = None
+    try:
+        backing_vol_obj = volume.vs.fh
+    except (AttributeError, TypeError):
+        return False
+
+    if backing_vol_obj in context.volume_region_map:
```

Comment:
```suggestion
    if backing_vol_obj not in context.volume_region_map:
        return False
```
same with the indentation allergy above.